engine/table/src/main/java/io/deephaven/engine/table/impl/BaseTable.java

```diff
@@ -630,6 +630,9 @@ public final void notifyListeners(final TableUpdate update) {
             Assert.assertion(getRowSet().sizePrev() == 0 || getRowSet().lastRowKeyPrev() < update.added().firstRowKey(),
                     "getRowSet().lastRowKeyPrev() < update.added().firstRowKey()");
         }
+        if (isStream()) {
```

Comment:
While you're in here, look at the `isFlat()` checks. `build()` should be `getRowSet()`.